pkg/security/secl/rules/policy.go

```diff
@@ -23,24 +23,34 @@ type PolicyDef struct {
 
 // Policy represents a policy file which is composed of a list of rules and macros
 type Policy struct {
-	Name    string
-	Source  string
-	Version string
-	Rules   []*RuleDefinition
-	Macros  []*MacroDefinition
+	Name               string
+	Source             string
+	Version            string
+	Rules              []*RuleDefinition
+	Macros             []*MacroDefinition
+	RuleSetTaggedRules map[eval.RuleSetTagValue][]*RuleDefinition
```

Comment:
I'm not sure the modification is required and I wouldn't change the policy definition. We could simply dispatch the rule according to its tag in the EvaluationSet.LoadPolicies:

```
def (e *EvaluationSet) LoadPolicies() {
  var defs map[string]RuleDefinitions

  for _, r := range policy.Rules {
     tag := r.GetTag("ruleset")
     defs[tag] = append(..., r)
  }

  for tag, rules := range defs{
    ruleSets[tag].AddRules(rules)
  }
}
```

This should reduce the number of modification introduced

Reply:
right!